### ISSUE TITLE
Vertico: adapt to consult recent API change

### ISSUE THREAD

**Issue by bixuanzju** [hlissner/doom-emacs#6277]
### What did you expect to happen?

When I press `SPC b b`, no errors in minibuffer.

### What actually happened?

When I press `SPC b b`, minibuffer shows
```
consult--read: No preview, the :state function protocol changed: (wrong-type-argument stringp setup)
consult--read: No preview, the :state function protocol changed: (wrong-type-argument stringp preview)
consult--read: No preview, the :state function protocol changed: (wrong-type-argument stringp exit)
consult--read: No preview, the :state function protocol changed: (wrong-type-argument stringp return)
```

### Describe your attempts to resolve the issue

API change in https://github.com/minad/consult/pull/546, I think we need to adapt this function `+vertico--workspace-buffer-state`

### Steps to reproduce

1. Open Emacs
2. `SPC b b`

### System Information

That website is blocked in my company.

**Comment by minad** [comment]
@iyefrat prepare a fix for this issue in #6273.

**Comment by hlissner** [comment]
This issue should now be resolved. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!
